packages/mobile/src/send/Send.tsx

```diff
@@ -168,50 +174,66 @@ class Send extends React.Component<Props, State> {
   }
 
   componentDidUpdate(prevProps: Props) {
-    const { recentRecipients, allRecipients } = this.props
+    const { recentRecipients, phoneRecipients, valoraRecipients } = this.props
 
+    // TODO: also check valoraRecipients
     if (
       recentRecipients !== prevProps.recentRecipients ||
-      allRecipients !== prevProps.allRecipients
+      phoneRecipients !== prevProps.phoneRecipients ||
+      valoraRecipients !== prevProps.valoraRecipients
     ) {
       this.createRecipientSearchFilters(
         recentRecipients !== prevProps.recentRecipients,
-        allRecipients !== prevProps.allRecipients
+        phoneRecipients !== prevProps.phoneRecipients,
+        valoraRecipients !== prevProps.valoraRecipients
       )
       // Clear search when recipients change to avoid tricky states
       this.onSearchQueryChanged('')
     }
   }
 
-  createRecipientSearchFilters = (updateRecentFilter: boolean, updateAllFilter: boolean) => {
+  createRecipientSearchFilters = (
+    updateRecentFilter: boolean,
+    updatePhoneFilter: boolean,
+    updateValoraFilter: boolean
+  ) => {
     // To improve search performance, we use these filter factories which pre-process the
     // recipient lists to improve search performance
     if (updateRecentFilter) {
       this.recentRecipientsFilter = filterRecipientFactory(this.props.recentRecipients, false)
     }
-    if (updateAllFilter) {
-      this.allRecipientsFilter = filterRecipientFactory(
-        Object.values(this.props.allRecipients),
+    if (updatePhoneFilter) {
+      this.phoneRecipientsFilter = filterRecipientFactory(
+        Object.values(this.props.phoneRecipients),
         true,
         this.props.matchedContacts
       )
     }
+    if (updateValoraFilter) {
+      this.valoraRecipientsFilter = filterRecipientFactory(
+        Object.values(this.props.valoraRecipients),
+        true
+      )
+    }
+
+    // TODO: update valoraRecipients filter
```

Comment:
Is this TODO still relevant?

Reply:
I don't think we need to do this any time soon but if we notice any slowdowns while searching really large contact lists we could try implementing it in sqlite or something similar. Maybe we can start by adding some performance logs to measure.